src/components/Footer.tsx

```diff
@@ -0,0 +1,77 @@
+import classNames from 'classnames';
+import { useMemo } from 'react';
+import { FilterType } from '../types/FilterType';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  filterBy: string;
+  setFilterBy: (value: FilterType) => void;
+  deleteCompletedTodos: () => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filterBy,
+  setFilterBy,
+  deleteCompletedTodos,
+}) => {
+  const activeTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed)
+  ), [todos]);
+
+  const completedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed)
+  ), [todos]);
```

Comment:
this constant contains a link on the array, this link won't change if only you are rewriting the array in total.